CONTRIBUTING.md

```diff
@@ -164,3 +187,39 @@ Do your best to follow the clean code guidelines.
 - Don’t return an error code, throw an exception instead.
 
 Ref: Book: Clean Code by Robert C. Martin (Uncle Bob)
+
+## Compilation and Installation
+This guide is helpful to compile and install Nmstate from source, for installing stable release or other installation methods, pleaser refer to [Nmstate installation guide](https://nmstate.io/user/install.md)
+
+### Dependencies
+Nmstate requires NetworkManager 1.26 or later version installed and started.
+
+In order to support specific capabilities, additional packages are required:
+
+OpenVSwitch support through the NM provider requires NetworkManager-ovs.
+
+Extended ovsdb support requires openvswitch-2.11 and python3-openvswitch2.11 or greater.
+
+Team interface support through the NM provider requires NetworkManager-team
+
+### Clone the source
+```
+git clone https://github.com/nmstate/nmstate.git
+cd nmstate
+```
+
+### Compilation 
+```
+#Check that the system installed the official compiler for the rust programming language
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+#Compile
+make
+```
+
+### Installation
+```
+sudo make install
+```
+
+__Note__: In order to report current state, installing Nmstate as a non-root user should be enough. For change support, install the package as root.
```

Comment:
I do not think this comment is correct. Installing in most situation must be do a root. I think you are mixing executing with installing.

A better note could be: `In order to report current state, running nmstatectl as a non-root user should be enough. For apply support, run nmstatectl as root.`